File: FitBook.Model/Responses/Payments/MembershipPaymentResponse.cs
using FitBook.Model.Enums;

namespace FitBook.Model.Responses.Payments;

public class MembershipPaymentResponse : IEntityResponse
{
    public int Id { get; set; }
    public decimal Amount { get; set; }
    public string Currency { get; set; } = string.Empty;
    public string PaymentProvider { get; set; } = string.Empty;
    public PaymentStatus Status { get; set; }
    public DateTime? PaidAtUtc { get; set; }
    public DateTime? RefundedAtUtc { get; set; }
    public decimal? RefundAmount { get; set; }
    public int UserMembershipId { get; set; }
    public int UserAccountId { get; set; }
    public DateTime CreatedAtUtc { get; set; }
    public DateTime? UpdatedAtUtc { get; set; }
}

File: UI/fitbook_mobile/lib/models/responses/membership_payment_response.dart
import 'package:json_annotation/json_annotation.dart';

import '../enums/payment_status.dart';

part 'membership_payment_response.g.dart';

@JsonSerializable()
class MembershipPaymentResponse {
  MembershipPaymentResponse({
    required this.id,
    required this.amount,
    required this.currency,
    required this.paymentProvider,
    required this.status,
    this.paidAtUtc,
    this.refundedAtUtc,
    this.refundAmount,
    required this.userMembershipId,
    required this.userAccountId,
    required this.createdAtUtc,
    this.updatedAtUtc,
  });

  final int id;
  final double amount;
  final String currency;
  final String paymentProvider;
  final PaymentStatus status;
  final DateTime? paidAtUtc;
  final DateTime? refundedAtUtc;
  final double? refundAmount;
  final int userMembershipId;
  final int userAccountId;
  final DateTime createdAtUtc;
  final DateTime? updatedAtUtc;

  factory MembershipPaymentResponse.fromJson(Map<String, dynamic> json) =>
      _$MembershipPaymentResponseFromJson(json);

  Map<String, dynamic> toJson() => _$MembershipPaymentResponseToJson(this);
}
